proto/src/lib.rs

```diff
@@ -133,4 +136,27 @@ pub mod health {
     }
 }
 
+#[cfg(all(feature = "protobuf-codec", not(feature = "prost-codec")))]
+#[allow(bare_trait_objects)]
+#[allow(renamed_and_removed_lints)]
+pub mod testing {
+    include!(concat!(env!("OUT_DIR"), "/testing/mod.rs"));
+}
+
+#[cfg(all(feature = "protobuf-codec", not(feature = "prost-codec")))]
```

Comment:
I think only one feature should be activated.

Reply:
Unfortunately Cargo will build with multiple features, even if they're not usually used together. This happened when running `cargo test --all`. Unfortunately Cargo doesn't have any way of finding out why certain features are used. That might be due to the mistake in the Cargo.toml you noted above. I'll try it out....